mysql/resource_user.go

```diff
@@ -257,8 +309,25 @@ func ReadUser(ctx context.Context, d *schema.ResourceData, meta interface{}) dia
 			d.Set("user", m[1])
 			d.Set("host", m[2])
 			d.Set("auth_plugin", m[3])
-			d.Set("auth_string_hashed", m[4])
 			d.Set("tls_option", m[5])
+
+			if m[3] == "aad_auth" {
+				// AADGroup:98e61c8d-e104-4f8c-b1a6-7ae873617fe6:upn:Doe_Family_Group
+				// AADUser:98e61c8d-e104-4f8c-b1a6-7ae873617fe6:upn:little.johny@does.onmicrosoft.com
+				// AADSP:98e61c8d-e104-4f8c-b1a6-7ae873617fe6:upn:mysqlUserName
+				parts := strings.Split(m[4], ":")
+				if parts[0] == "AADSP" {
+					// service principals are referenced by UUID only
```

Comment:
Can something like this happen? Someone will use UUID-only name and it will be user/group or vice versa. That would mean we'd have a persistent diff here.

Reply:
For users, this is not possible, as user has to be referenced as UPN (`user@tenant`). So even if username is any guid, upn is then `guid@tenantname.onmicrosoft.com` for example.

For groups - yes, this is possible problem. And I really don't know how to solve this smoothly. I will send commit with changing aad_identity to block type with fields type and identity, which is best solution I have found